src/src_roughness_optimization.py

```diff
@@ -148,232 +286,431 @@ def update_rating_curve(fim_directory, water_edge_median_df, htable_path, huc, b
+        )  # create a new deque that will be used to populate all relevant downstream hydroids
+        vert_count = 0
+        branch_count += 1
         while Q:
             q = Q.popleft()
             if q not in visited:
-                df_input_htable.loc[df_input_htable.HydroID==q,'route_count'] = vert_count # assign var with flow order ranking
-                df_input_htable.loc[df_input_htable.HydroID==q,'branch_id'] = branch_count # assign var with current branch id
+                df_input_htable.loc[
+                    df_input_htable.HydroID == q, 'route_count'
+                ] = vert_count  # assign var with flow order ranking
+                df_input_htable.loc[
+                    df_input_htable.HydroID == q, 'branch_id'
+                ] = branch_count  # assign var with current branch id
                 vert_count += 1
                 visited.add(q)
-                nextid = df_input_htable.loc[q,'NextDownID'] # find the id for the next downstream hydroid
-                order = df_input_htable.loc[q,'order_'] # find the streamorder for the current hydroid
-            
+                # find the id for the next downstream hydroid
+                nextid = df_input_htable.loc[q, 'NextDownID']
+                order = df_input_htable.loc[q, 'order_']  # find the streamorder for the current hydroid
+
                 if nextid not in visited and nextid in df_input_htable.HydroID:
-                    check_confluence = (df_input_htable.NextDownID == nextid).sum() > 1 # check if the NextDownID is referenced by more than one hydroid (>1 means this is a confluence)
-                    nextorder = df_input_htable.loc[nextid,'order_'] # find the streamorder for the next downstream hydroid
-                    if nextorder > order and check_confluence == True: # check if the nextdownid streamorder is greater than the current hydroid order and the nextdownid is a confluence (more than 1 upstream hydroid draining to it)
-                        branch_heads.append(nextid) # found a terminal point in the network (append to branch_heads for second pass)
-                        continue # if above conditions are True than stop traversing downstream and move on to next starting hydroid
+                    # check if the NextDownID is referenced by more than one hydroid
+                    # (>1 means this is a confluence)
+                    check_confluence = (df_input_htable.NextDownID == nextid).sum() > 1
+                    nextorder = df_input_htable.loc[
+                        nextid, 'order_'
+                    ]  # find the streamorder for the next downstream hydroid
+                    # check if the nextdownid streamorder is greater than the current hydroid order and the
+                    # nextdownid is a confluence (more than 1 upstream hydroid draining to it)
+                    if nextorder > order and check_confluence == True:
```

Comment:
@CarsonPruitt-NOAA @RyanSpies-NOAA @mluck - I just confirmed that this line was indeed the cause of the difference in hashes we experienced earlier. For reference:

`if nextorder > order and check_confluence is True:` was the previous version that altered the behavior.